src/main/java/ca/on/oicr/pineryreports/reports/impl/OctaneCountsReport.java

```diff
@@ -0,0 +1,487 @@
+    caseNumbers.add(new Count("Tumor Tissue RNA Transferred", countUniqueIdentities(rnaAliquotsFromSlides, allOctaneSamplesById)));
+    caseNumbers.add(new Count("Buffy Coat Rec'd", countUniqueNewIdentities(buffyCoats, allOctaneSamplesById)));
+    caseNumbers.add(new Count("Buffy Coat Extracted", getUniqueIdentities(stocksFromBuffy, allOctaneSamplesById).size()));
+    caseNumbers.add(new Count("Buffy Coat Transferred", countUniqueIdentities(dnaAliquotsFromBuffy, allOctaneSamplesById)));
+    caseNumbers.add(new Count("Plasma Rec'd", countUniqueNewIdentities(plasma, allOctaneSamplesById)));
+    caseNumbers.add(new Count("ctDNA Plasma Rec'd", countUniqueNewIdentities(ctDnaPlasma, allOctaneSamplesById)));
+    
+    
+    // Sample numbers
+    sampleNumbers.add(new Count("Tumor Tissue Rec'd", countSlidesReceived(slides)));
+    sampleNumbers.add(new Count("Tumor Tissue Extracted", stockDnaFromSlides.size()));
+    sampleNumbers.add(new Count("Tumor Tissue DNA Transferred", dnaAliquots.size()));
+    sampleNumbers.add(new Count("Tumor Tissue RNA Transferred", rnaAliquots.size()));
+    sampleNumbers.add(new Count("Buffy Coat Rec'd", buffyCoats.size()));
+    sampleNumbers.add(new Count("Buffy Coat Extracted", stocksFromBuffy.size()));
+    sampleNumbers.add(new Count("Plasma Rec'd", plasma.size()));
+    sampleNumbers.add(new Count("Plasma Extracted", ctDnaPlasma.size()));
+  }
+  
+  private void addSlidesRemainingCount(SampleDto slide) {
+    Integer slides = getIntAttribute(ATTR_SLIDES, slide, 0);
+    Integer discards = getIntAttribute(ATTR_DISCARDS, slide, 0);
+    AttributeDto attr = new AttributeDto();
+    attr.setName(ATTR_REMAINING);
+    attr.setValue(Integer.toString(slides - discards));
+    slide.getAttributes().add(attr);
+  }
+  
+  private void calculateConsumedSlides(Collection<SampleDto> samples, Map<String, SampleDto> potentialParents) {
+    samples.stream()
+        .filter(sam -> getAttribute(ATTR_CONSUMED, sam) != null).forEach(sam -> {
+          sam.getParents().forEach(ref -> {
+            SampleDto parent = potentialParents.get(ref.getId());
+            AttributeDto remainingAttr = parent.getAttributes().stream().filter(attr -> ATTR_REMAINING.equals(attr.getName())).findFirst().get();
+            Integer remaining = Integer.parseInt(remainingAttr.getValue());
+            Integer consumed = Integer.parseInt(getAttribute(ATTR_CONSUMED, sam));
+            remainingAttr.setValue(Integer.toString(remaining - consumed));
+          });
+        });
+  }
+  
+  private int countSlidesRemaining(Collection<SampleDto> samples) {
+    return samples.stream()
+        .mapToInt(s -> getIntAttribute(ATTR_REMAINING, s))
+        .sum();
+  }
+  
+  private int countSlidesReceived(Collection<SampleDto> samples) {
+    return samples.stream()
+        .mapToInt(s -> getIntAttribute(ATTR_SLIDES, s))
+        .sum();
+  }
+  
+  private static Predicate<SampleDto> byStain(String stain) {
+    return slide -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement and then remove useless return keyword [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)